Lib/tarfile.py

```diff
@@ -2447,20 +2447,52 @@ def is_tarfile(name):
 
 def main():
     import argparse
+    compressions = {
+        # gz
+        '.gz': 'gz',
+        '.tgz': 'gz',
+        # xz
+        '.xz': 'xz',
+        '.txz': 'xz',
+        # bz2
+        '.bz2': 'bz2',
+        '.tbz': 'bz2',
+        '.tbz2': 'bz2',
+        '.tb2': 'bz2',
+    }
+    tar_files = []
+    tar_name = None
+    tar_mode = None
 
     description = 'A simple command-line interface for tarfile module.'
-    parser = argparse.ArgumentParser(description=description)
+    epilog= ' '.join([
```

Comment:
Tip: the Python compiler will automatically merge string literals that follow each other.

```python

epilog = ('Beginning of text, '
          'followed by more text.')
```

Here you also need dynamic text replacement; a simple way to do that is to use an f-string:

```python

available_compressions = ', '.join(compressions)
epilog = f'final extension (one of {available_compressions})'

Reply:
Good tips, (I guess I am still stuck in python 2 mode). Also reworked tarfile.main & zipfile.main docstrings to make more grammatical, (and better spelt). 

Also added note to zipfile epilogue that only a single compress can be used at invocation but later adds can use a different compression.

Reply:
Also only offer available compression methods in zipfile.main options.

Reply:
And only mention available compression methods as currently available.